### PR TITLE
Chat Channel List Scrolling Issue

### PR DESCRIPTION
**Pull request by jitendra-simform**
### 🔗 Issue Link
https://docs.google.com/document/d/1ufYv5_H7LxeTPWh7TwyaQnwSSNlDWQ2cJhpGFB6FAww/edit

### 🎯  Fixed Bugs
58. Performance on the “Chat" screen:
Jitter, stutter when scrolling
50 Group's avatar flashes multiple times when user open chat

### 📝 Changes

Shimmer effect added while chat group avatar loading
Now chat group channel avatars loaded in background queue

### CODE REVIEW

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift [hunk 1]
```diff
@@ -159,7 +213,7 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
         
         var combinedImage: UIImage?
         
-        let imageProcessor = components.imageProcessor
+        let imageProcessor = imageProcessor
```
Comment (mdhanif-simformsolutions) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift:217]: Can we remove this duplicate imageProcessor variable ?

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift [hunk 2]
```diff
@@ -25,13 +31,27 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
...
+        embed(shimmerView)
+        shimmerView.isSkeletonable = true
     }
 
     override open func updateContent() {
+        loadIntoAvatarImageView(from: nil, placeholder: nil)
+        shimmerView.layer.cornerRadius = avatarCornerRadius
+        shimmerView.skeletonCornerRadius = Float(avatarCornerRadius)
+        shimmerView.showAnimatedGradientSkeleton()
+        shimmerView.isHidden = true
```
Comment (mdhanif-simformsolutions) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift:54]: As we are having `shimmerView.isHidden = true`  do we need show Animated Gradient here  `shimmerView.showAnimatedGradientSkeleton()` ?

**Review by mdhanif-simformsolutions — COMMENTED** [review]
@jitendra-simform Please have a look on the comments.

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift [hunk 3]
```diff
@@ -101,12 +134,33 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
         
         // We show a combination of at max 4 images combined
         urls = Array(urls.prefix(maxNumberOfImagesInCombinedAvatar))
-        
+        let customKeyForCache = urls.compactMap({ $0?.lastPathComponent}).joined()
+        // Checked if avatar already cached
+        if let imageContainer = ImagePipeline.shared.cache.cachedImage(for: customKeyForCache), let cachedUrlCount = imageContainer.userInfo["count"] as? Int {
+            if urls.count == cachedUrlCount{
+                loadIntoAvatarImageView(from: nil, placeholder: imageContainer.image)
```
Comment (mdhanif-simformsolutions) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift:141]: Check for character count linting here. 
Also update `if urls.count == cachedUrlCount` with the above if let only.

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift (hunk 2 above)
Comment (jitendra-simform) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift:54]: @mdhanif-simformsolutions  Yes , if we don't start shimmer effect here , shimmer effect will not working as aspected, Thanks

File: Sources/StreamChatUI/xibs/SkeletonViewSetting/SkeletonView+Configure.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
+//
+//  SkeletonView+Configure.swift
+//  StreamChat
+//
+//  Created by Jitendra Sharma on 15/04/22.
+//  Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
```
Comment (ajay-simform) [inline Sources/StreamChatUI/xibs/SkeletonViewSetting/SkeletonView+Configure.swift:9]: @jitendra-simform  Please remove `import Foundation`

**Pull request merged**